repos/system_upgrade/el8toel9/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM
+
+
+report_summary = (
+    'NIS components (ypserv, ypbind, yp-tools) will no longer be available in RHEL-9.'
+    ' Technology behind those packages is based an outdated design patterns no longer'
+    ' considered secure. There is no direct alternative with fully compatible features.'
+)
+
+report_hint = (
+    'The alternatives are LDAP and for some use cases Kerberos. Migrating to IPA'
+    ' is also a solution.'
+)
+
+report_link_url = 'https://access.redhat.com/solutions/5991271'
+
+PACKAGES_NAMES = ('ypserv', 'ypbind')
+YPBIND_CONF_FILE = '/etc/yp.conf'
+YPSERV_DIR_PATH = '/var/yp'
+YPSERV_DEFAULT_FILES = ('binding', 'Makefile', 'nicknames')
+
+def report_nis_installed(packages):
+    """
+    Create the report if any of NIS packages (RH signed)
+    is installed and configured.
+
+    Should notify user about present NIS compnent package
+    installation, warn them about discontinuation, and
+    redirect them to online documentation for possible
+    alternatives.
+    """
+
+    report_content = [
+        reporting.Title('NIS component has been detected on your system'),
+        reporting.Summary(report_summary),
+        reporting.Severity(reporting.Severity.MEDIUM),
+        reporting.Tags([reporting.Tags.SERVICES]),
+        reporting.ExternalLink(title='RHEL 9 (NIS) discontinuation',
+                                url=report_link_url),
+        reporting.Remediation(hint=report_hint),
+    ]
+
+    for package in packages:
+        reporting.RelatedResource('package', package)
+
+    reporting.create_report(report_content)
+
+def report_installed_packages():
+    """
+    Create reports according to detected NIS packages
+    and their configuration.
+    """
+    installed_packages = []
+    nis_configured = False
+
+    for package in PACKAGES_NAMES:
+        if has_package(InstalledRedHatSignedRPM, package):
+            installed_packages.append(package)
+
+    if 'ypbind' in installed_packages:
+        nis_configured = check_client_configuration()
+
+    if 'ypserv' in installed_packages:
+        nis_configured = True if check_server_configuration() else nis_configured
+
+    if installed_packages and nis_configured:
+        report_nis_installed(installed_packages)
+
+def check_client_configuration():
+    """
+    Check for any significant ypbind configuration lines in .conf file.
+    """
+    if not os.path.isfile(YPBIND_CONF_FILE):
```

Comment:
in the check phase, actors must not communicate with the system. split it into 2 actors (aka scanner, checker) where the scanner gets all required data, propagates them via a msg, and the check actor does additoinal actions.

Reply:
New nisscaner actor new provides this information for nischeck.